security-proxy/src/main/java/org/georchestra/security/HeadersManagementStrategy.java

```diff
@@ -303,59 +313,114 @@ public synchronized void copyResponseHeaders(HttpServletRequest originalRequest,
         }
 
         Header[] cookieHeaders = proxyResponse.getHeaders(HeaderNames.SET_COOKIE_ID);
-        if (cookieHeaders != null) {
+        if (cookieHeaders.length > 0) {
             handleResponseCookies(originalRequestURI, finalResponse, cookieHeaders, session);
         }
     }
 
-    private void handleResponseCookies(String originalRequestURI, HttpServletResponse finalResponse, Header[] headers,
+    /**
+     * Manages the "Set-Cookie" headers coming from security-proxified webapps.
+     * 
+     * If the cookie is a JSESSIONID, then we need to keep it into a map stored into
+     * the current user's session, but not transmit it to the client, as there is
+     * probably already a sessionid between the client and the SP. It will be reused
+     * for later requests to the same proxified webapp.
+     * 
+     * In the other cases, it is probably safe to transmit it to the client, but we
+     * force the path to the currently proxified webapp.
+     *
+     * If the cookie is transmitted to the client, the Path can be mangled to limit
+     * its scope to the proxified webapp.
+     *
+     * @param originalRequestURI the request URI being made. We need it to force the
+     *                           path of the cookies
+     * @param finalResponse      the actual response sent to the client
+     * @param headers            the array of "Set-Cookie" headers
+     * @param session            the current user's session
+     */
+    @VisibleForTesting
+    public void handleResponseCookies(String originalRequestURI, HttpServletResponse finalResponse, Header[] headers,
             HttpSession session) {
-        String originalPath = originalRequestURI.split("/")[0];
+        String overridenPath = null;
+        try {
+            overridenPath = originalRequestURI.split("/")[1];
+        } catch (Exception e) {
+        }
```

Comment:
Can't get this part, why are we silently continue ? maybe at least log ? it means that `overridenPath` will be null. Could lead to NPE.

Reply:
This should never happen, because if we got into this part of the code, it is due to a call to `<scheme>://<my georchestra>/<webapp` declared in the `targets-mapping.properties`. The default request mapping is handled before in the code:

https://github.com/georchestra/georchestra/blob/19.04/security-proxy/src/main/java/org/georchestra/security/Proxy.java#L352-L361

I think it should be even safe to throw an exception here actually